lib/eddystone-beacon-scanner.js

```diff
@@ -24,47 +32,188 @@ EddystoneBeaconScanner.prototype.startScanning = function(allowDuplicates) {
+  inRange[id].lastSeen = Date.now();
+};
+
+EddystoneBeaconScanner.prototype.parseBeacon = function(peripheral) {
+  var frameType = this.getFrameType(peripheral.advertisement.serviceData[0].data[0]);
+
+  switch(frameType) {
+      case 'uid':
+        return this.parseUidFrame(peripheral);
+        break;
+
+      case 'url':
+        return this.parseUrlFrame(peripheral);
+        break;
+
+      case 'tlm':
+        return this.parseTlmFrame(peripheral);
+        break;
+    }
+};
+
+EddystoneBeaconScanner.prototype.parseUidFrame = function(peripheral) {
   var data = peripheral.advertisement.serviceData[0].data;
 
   return {
-    type: 'url',
+    frameType: this.getFrameType(data.readUInt8(0)),
     txPower: data.readInt8(1),
-    url: uriDecode(data.slice(2)),
-    rssi: peripheral.rssi
+    namespace: data.slice(2, 12).toString('hex'),
+    instance: data.slice(12, 18).toString('hex'),
+    rssi: peripheral.rssi,
+    distance: this.calculateDistance(data.readUInt8(1), peripheral.rssi)
   };
 };
 
+EddystoneBeaconScanner.prototype.parseUrlFrame = function(peripheral) {
+  var data = peripheral.advertisement.serviceData[0].data;
+
+  return {
+    frameType: this.getFrameType(data.readUInt8(0)),
+    txPower: data.readInt8(1),
+    url: urlDecode(data.slice(2)),
+    rssi: peripheral.rssi,
+    distance: this.calculateDistance(data.readUInt8(1), peripheral.rssi)
+  };
+};
+
+EddystoneBeaconScanner.prototype.parseTlmFrame = function(peripheral) {
+  var data = peripheral.advertisement.serviceData[0].data;
+  console.log('TLM Frame', data);
+};
+
+EddystoneBeaconScanner.prototype.getFrameType = function(frameType) {
+  switch(frameType) {
+    case UID_FRAME_TYPE: 
+      return 'uid';
+      break;
+    case URL_FRAME_TYPE:
+      return 'url'
+      break;
+    case TLM_FRAME_TYPE:
+      return 'tlm'
+      break;
+    default:
+      throw new Error('Invalid Frametype');
+      break;
+  }
+};
+
+EddystoneBeaconScanner.prototype.calculateDistance = function(txPower, rssi) {
+  // return Math.pow(10, ((txPower - rssi) - 41) / 20.0);
+
+  /* 
+  The distance is an estimate, and the accuracy will vary based
+  on the accuracy of the samples provided and the wireless
+  characteristics of your environment.  It will most likely only
+  be accurate to a few feet, as triangulation is not used.
+  However, it does provide some entertainment value. 
+  */
+
+  var average = 1.0;  // must be outside value range (-0dBm to -127dBm)
+
+  // number of samples in the average
+  var samples = 20;
+
+  // sample #1 from environment (rssi -45dBm = distance 1.5ft)
+  var r1 = -45;
+  var d1 = 1.5;
+
+  // sample #2 from environment (rssi -75dBm = distance 20ft)
+  var r2 = -75;
+  var d2 = 20;
+
+  /* 
+  constant to account for loss due to reflection, polzarization, etc
+  n will be ~2 for an open space
+  n will be ~3 for "typical" environment
+  n will be ~4 for a cluttered industrial site 
+  */
+
+  var n = (r2 - r1) / (10 * Math.log10(d1 / d2));
```

Comment:
I'm wondering if we should also stick to the same algorithm used in the google/eddystone repo examples for now.